### PR TITLE
PN-3820-timeline-tests-augmented-and-refactored

### PR DESCRIPTION
**Pull request by clombardi**
## Short description
Added tests regarding the separated texts for multi-recipient notifications, and for the added logic in parseNotificationDetail / populateMacroSteps. Refactored the test suite to avoid making the text mocks even more complex.

## List of changes proposed in this pull request
- New, refactored and augmented test suit for `notification.utility.ts` in pn-commons.
- Some cosmetic changes and code cleaning in `notification.utility.ts` and other code that regards timeline texts.

## How to test
Run `notification.utility.test.ts` in pn-commons.

### CODE REVIEW

File: packages/pn-commons/src/types/NotificationDetail.ts [hunk 1]
```diff
@@ -320,6 +325,12 @@ export enum DigitalDomicileType {
   APPIO = 'APPIO', // PN-2068
 }
 
+export enum DigitalDomicileTypeForCourtesyMessageOnly {
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/types/NotificationDetail.ts:328]: I would integrate this into interface DomicileType

File: packages/pn-commons/src/utils/__test__/notification.utility.test.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import {
   TimelineCategory,
   NotificationStatus,
 } from '../../types';
-import { NotificationDeliveryMode, NotificationStatusHistory, ResponseStatus } from '../../types/NotificationDetail';
+import { DigitalDomicileTypeForCourtesyMessageOnly, DigitalWorkflowDetails, INotificationDetailTimeline, NotificationDeliveryMode, NotificationDetail, NotificationDetailRecipient, NotificationStatusHistory, RecipientType, ResponseStatus } from '../../types/NotificationDetail';
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/utils/__test__/notification.utility.test.ts:14]: Merge this import with the above one and add missing imports to src/types/index.ts

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-commons/src/types/NotificationDetail.ts (hunk 1 above)
Comment (clombardi) [reply packages/pn-commons/src/types/NotificationDetail.ts:328]: OK, I feared about possible conflict with the manual notification creation about the digital domicile of each recipient, but now I see that the type PEC is hardcoded. Will do as suggested.

File: packages/pn-commons/src/utils/__test__/notification.utility.test.ts (hunk 2 above)
Comment (clombardi) [reply packages/pn-commons/src/utils/__test__/notification.utility.test.ts]: Done.

File: packages/pn-commons/src/types/NotificationDetail.ts (hunk 1 above)
Comment (clombardi) [reply packages/pn-commons/src/types/NotificationDetail.ts]: Done.

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**